components/Editor/index.tsx

```diff
@@ -0,0 +1,230 @@
+            <MenuItem onAction={() => commands.addTableColumnAfter()}>Add Column Right</MenuItem>
+            <MenuItem onAction={() => commands.deleteTableColumn()}>Delete Column</MenuItem>
+            <Separator />
+            <MenuItem onAction={() => commands.addTableRowBefore()}>Add Row Above</MenuItem>
+            <MenuItem onAction={() => commands.addTableRowAfter()}>Add Row Below</MenuItem>
+            <MenuItem onAction={() => commands.deleteTableRow()}>Delete Row</MenuItem>
+            <Separator />
+            <MenuItem onAction={() => commands.deleteTable()}>Delete Table</MenuItem>
+          </Menu>
+        </Popover>
+      </MenuTrigger>
+    </Group>
+  )
+}
+
+
+const EditorToolbar = () => {
+  const chain = useChainedCommands();
+  const active = useActive();
+
+  return (
+    <Toolbar aria-label="Editor Tools">
+      <Group aria-label="Style">
+        <ToggleButton
+          aria-label="Bold"
+          onPress={() => {
+            chain
+              .toggleBold()
+              .focus()
+              .run();
+          }}
+          isSelected={active.bold()}
+        >
+          <DmpIcon icon="format_bold" />
+        </ToggleButton>
+        <ToggleButton
+          aria-label="Italic"
+          onPress={() => {
+            chain
+              .toggleItalic()
+              .focus()
+              .run();
+          }}
+          isSelected={active.italic()}
+        >
+          <DmpIcon icon="format_italic" />
+        </ToggleButton>
+        <ToggleButton
+          aria-label="Underline"
+          onPress={() => {
+            chain
+              .toggleUnderline()
+              .focus()
+              .run();
+          }}
+          isSelected={active.underline()}
+        >
+          <DmpIcon icon="format_underlined" />
+        </ToggleButton>
+      </Group>
+
+      <Separator orientation="vertical" />
+
+      <Group aria-label="Lists">
+        <ToggleButton
+          aria-label="Bullet List"
+          onPress={() => {
+            chain
+              .toggleBulletList()
+              .focus()
+              .run();
+          }}
+          isSelected={active.bulletList()}
+        >
+          <DmpIcon icon="format_list_bulleted" />
+        </ToggleButton>
+
+        <ToggleButton
+          aria-label="Number List"
+          onPress={() => {
+            chain
+              .toggleOrderedList()
+              .focus()
+              .run();
+          }}
+          isSelected={active.orderedList()}
+        >
+          <DmpIcon icon="format_list_numbered" />
+        </ToggleButton>
+      </Group>
+
+      <Separator orientation="vertical" />
+
+      <TableGroup />
+
+    </Toolbar>
+  )
+}
+
+
+interface DmpEditorProps {
+  content: string;
+  setContent: (newContent: string) => void;
+}
+
+export function DmpEditor({content, setContent}: DmpEditorProps) {
+  const { manager, state, setState } = useRemirror({
+    extensions: () => [
+      new BoldExtension(),
```

Comment:
L188 - @andrewebdev I see errors that at least one argument needs to be passed to BoldExtension, BulletListExtension, TableExtension, and AnnotationExtension. You can probably just pass an empty object, like "{}"

Reply:
I'll make those changes, but not sure why they were not flagged in my linter.
I run `npx eslint '**/*.ts'` to check for things like this, but it didn't flag anything.

Reply:
@andrewebdev this is a Typescript bug that was highlighted in VSCode during my review of your code.